src/raw-components.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useContext } from "react";
+import ReactDOMServer from "react-dom/server";
+import { tags } from "./tags";
+
+type Contexts = React.Context<unknown>[];
+export const MetaContext = React.createContext<Contexts>([]);
+
+type JSXInEl = JSX.IntrinsicElements;
+
+type Props<Tag extends SupportedTag> = {
+  as: Tag;
+  renderer?: typeof ReactDOMServer.renderToStaticMarkup;
+  createElement?: typeof React.createElement;
+} & PropsOf<Tag>;
+
+export function RawHTMLWrapper<Tag extends SupportedTag>({
+  as,
+  children = [],
+  renderer = ReactDOMServer.renderToStaticMarkup,
+  createElement = React.createElement,
+  ...rest
+}: Props<Tag>) {
+  const contexts = useContext(MetaContext);
+  const values = contexts.map((c) => useContext(c));
```

Comment:
Under what situations could the list of contexts change? If they change between calls to this function then this map would break the rules of hooks.

Reply:
It does break the rules of hooks, but here's a quick test that shows it has no effect on the `useContext` hook (unlike `useEffect` and `useState`, etc. that are deeply affected) https://codesandbox.io/s/order-of-hooks-if6ld?file=/src/App.js

In our downstream uses this hasn't caused any bugs yet, but context hasn't really been changing